src/Service/EloquaApiClient.php

```diff
@@ -131,6 +131,18 @@ public function getAccessTokenByRefreshToken() {
       }
     }
     else {
+      // If both access and refresh tokens are expired/invalid use fallback
+      // method to generate access and refresh tokens using resource owner
+      // password grant authorization.
+      // Also make sure that auth fallback service exists i.e. module that
+      // holds this service is enabled.
+      if (\Drupal::hasService('eloqua_api_auth_fallback.commands')) {
```

Comment:
It would be better if this is Dependency Injected. See
[Dependency Injection if service exists](https://drupal.stackexchange.com/questions/273178/dependency-injection-if-service-exists)

Reply:
I tried to inject the auth fallback service using service provider approach however since there is dependency already added in auth fallback service for eloqua client, system ran into circular reference error with following message

> Symfony\Component\DependencyInjection\Exception\ServiceCircularReferenceException: Circular reference detected for service "eloqua_api_redux.client", path: "eloqua_api_redux.forms -> eloqua_api_redux.client -> eloqua_api_redux.auth_fallback_default". in Drupal\Component\DependencyInjection\Container->get() (line 141 of /var/www/rmld8/web/core/lib/Drupal/Component/DependencyInjection/Container.php).

Tried following approaches but still couldn't resolve this issue.
1. Lowering the dependency priorities
2. Declaring service to be loaded with lazy 

Below are some of the similar drupal org issues 
- https://www.drupal.org/project/domain_language/issues/3007609
- https://www.drupal.org/project/domain/issues/3005048
- https://www.drupal.org/project/amp/issues/3050965

Had to resort with [static loading of the auth fallback service](https://github.com/dakkusingh/eloqua_api_redux/pull/1/commits/db9c112b7b03869418e694dac1c35d57d36e982d#diff-bf1636aaa4e03b2e3f702b491e26a061R138). 


I have implemented default auth fallback service and overridden it with submodule service using service decoration pattern. I felt this approach is clean. Let me know if this fine else will revert to earlier implementation.

Reply:
> overridden it with submodule 
At a glance that looks quite neat as well.